helpdesk-maker-checker-service/src/main/java/org/symphonyoss/symphony/bots/helpdesk/makerchecker/model/check/AgentExternalCheck.java

```diff
@@ -110,24 +109,19 @@ private boolean hasAttachmentsInMessage(SymMessage message) {
   }
 
   @Override
-  public Set<SymMessage> makeApprovedMessages(MakerCheckerMessage makerCheckerMessage,
+  public SymAttachmentInfo getApprovedAttachment(MakerCheckerMessage makerCheckerMessage,
       SymMessage symMessage) {
     AttachmentMakerCheckerMessage checkerMessage = (AttachmentMakerCheckerMessage) makerCheckerMessage;
+    SymAttachmentInfo attachmentInfoApproved = new SymAttachmentInfo();
 
-    SymMessage acceptMessage = new SymMessage();
-    List<SymAttachmentInfo> attachmentInfoSet = new ArrayList<>();
     for(SymAttachmentInfo attachmentInfo: symMessage.getAttachments()) {
```

Comment:
Use Java8 streams with a filter rather than the for-each loop to make the code cleaner and immutable.